### PR TITLE
Admin crud

### LINKED ISSUE #6: Work on Users Node Components and create necessary user APIs to create, search, and find all users

### LINKED ISSUE #8: Update API’s User and Tuit requests as needed

### PR DESCRIPTION
**Pull request by aliu19**

### CODE REVIEW

File: controllers/UserController.ts [hunk 1]
```diff
@@ -89,14 +106,63 @@ export default class UserController implements UserControllerI {
...
+     * @param {Request} req Represents request from client, including path
+     * parameter uid identifying the primary key of the user to be removed
+     * @param {Response} res Represents response to client, including status
+     * on whether deleting a user was successful or not
+     */
+    deleteUser = (req: Request, res: Response) => {
+        // @ts-ignore
+        const profile = req.session['profile'];
+        if (profile) {
+            if (profile._id == req.params.uid) {
```
Comment (caijiah) [inline controllers/UserController.ts]: missing a '='. I can fix it.

File: controllers/UserController.ts [hunk 2]
```diff
@@ -89,14 +106,63 @@ export default class UserController implements UserControllerI {
...
+    /**
+     * Removes a user instance from the database
+     * @param {Request} req Represents request from client, including path
+     * parameter uid identifying the primary key of the user to be removed
+     * @param {Response} res Represents response to client, including status
+     * on whether deleting a user was successful or not
+     */
+    deleteUser = (req: Request, res: Response) => {
+        // @ts-ignore
+        const profile = req.session['profile'];
```
Comment (caijiah) [inline controllers/UserController.ts]: We might need implement this method in authController because of using session.

File: daos/TuitDao.ts [hunk 3]
```diff
@@ -106,4 +106,13 @@ export default class TuitDao implements TuitDaoI {
...
\ No newline at end of file
+
+    /**
+     * Uses TuitModel to retrieve tuit document(s) from tuits collection
+     * using the given tuit
+     * @param {string} tuit Tuit's tuit
+     * Promise to be notified when tuit(s) are retrieved from the database
+     */
+    searchByTuit = async (tuit: string): Promise<any> =>
+        TuitModel.find({tuit: {$regex: tuit}})
```
Comment (caijiah) [inline daos/TuitDao.ts]: Could you add `populate('postedBy').exec();`

File: controllers/TuitController.ts [hunk 4]
```diff
@@ -42,6 +42,7 @@ export default class TuitController implements TuitControllerI {
             app.get('/api/tuits', TuitController.tuitController.findAllTuits);
             app.get('/api/tuits/:tid', TuitController.tuitController.findTuitById);
             app.get('/api/users/:uid/tuits', TuitController.tuitController.findTuitsByUser);
+            app.get('/api/admin/:tuit/tuits', TuitController.tuitController.searchByTuit)
```
Comment (caijiah) [inline controllers/TuitController.ts:46]: Could you add new endpoint to the documentation for TuitController above?

**Review by caijiah — APPROVED** [review]
Look good!

**Review by averywpx — APPROVED** [review]

**Pull request merged**